### DISCUSSION TITLE
State-Driven Dynamic CSS is not useful to <teleport>

### DISCUSSION THREAD

**Discussion by Jveen-D** [vuejs/rfcs#407] [General]
When I don't use `<teleport>`, `State-Driven Dynamic CS` is very useful, but when I try to use` State-Driven Dynamic CSS` to `<teleport>`, I ran into a little problem. I have the same writing as the official website, but the CSS values do not take effect when I F12 to check the webpage. Below is the code snippet.
```vue
<template>
  <div id="test" class="w-full mt-8">
    <p>hello</p>
  </div>
</template>

<script lang="ts" setup>
const theme = {
  color: 'red'
}
</script>

<style lang="scss" scoped>
p {
  color: v-bind('theme.color');
}
</style>
```
Needless to say, this is the correct usage，but then look down.
```vue
<template>
  <teleport to="#test">
      <p>hello</p>
  </teleport>
</template>

<script setup lang="ts">
const theme = {
  color: 'red'
}
</script>

<style lang="scss" scoped>
p {
  color: v-bind('theme.color');
}
</style>
```
This is using `<teleport>`below is the performance of the page.
![image](https://user-images.githubusercontent.com/46897216/139403769-21fce82c-1872-4580-baa6-81f476355365.png)
Next, f12 checks the page
![image](https://user-images.githubusercontent.com/46897216/139404039-2aa9102a-2619-4ec5-9070-1618af6a58a5.png)
![image](https://user-images.githubusercontent.com/46897216/139404084-42a51b41-5577-45e9-97c7-c9e296180b40.png)
What's the problem? I don’t seem to find this in the documentation. Is it because I wrote it wrong? Please help me to check it.

**Comment by posva** [comment]
This is because of how css variables work: they depend of the tree structure.

Please, next time consider using the [forum](http://forum.vuejs.org/), the [Discord server](https://vue-land.js.org/) or [StackOverflow](http://stackoverflow.com/tags/vue.js) for questions. The rfcs repo is not for questions.
